### ISSUE TITLE
Sync address collector with CardDAV

### ISSUE THREAD

**Issue by enoch85** [owncloud/mail#1298]
It would be awesome if the new [address collector](https://github.com/owncloud/mail/pull/1276) would sync contacts with CardDAV so that I can use the collected addresses on mobile as well.

**Comment by ChristophWurst** [comment]
Yes, that would be a nice feature. The question is how and when we should add which information to which address book.
I can think of two possible ways:

1. Add contact automatically – some kind of default address book is used
2. Ask the user if she or he wants to add that person to a selectable address book

@jancborchardt do you have any idea what that could look like?

Technically, this could be tricky to implement as there's no such contacts integration for adding contacts AFAIK. That means we would either have to extend core for that or directly talk to the contacts/dav app. I'd prefer the first approach as the latter one causes app-dependencies.

**Comment by DeepDiver1975** [comment]
Adding an E-Mail+displayname to am addressbook is an explicit operation by the user from my pov.
This can be implemented as pure client side operation using an JS DAV library.

**Comment by jancborchardt** [comment]
Gmail does this automatically and adds people to a specific group (or address book, no idea how this is handled in the backend) called »Other contacts«.

We could create this group and add people there. Or make it a separate address book which is not enabled/synced by default.

From that logic (have the contacts available but don’t have them clutter up your Contacts & synced) our current approach with the separate DB is fine. We need to reinstate the feature that you can add people to your Contacts by clicking on their name, which would open an action menu with »Add to contacts« and »Write message«.

If they are in your contacts already, instead of »Add to contacts« we could have a »Show profile« which could show a small card with avatar and other info (like company, phone nr etc) as a popover, like Twitter / Facebook do.

**Comment by DeepDiver1975** [comment]
> Gmail does this automatically and adds people to a specific group

they can easily do this since they have full control over the full backend stack - we don't.

**Comment by DeepDiver1975** [comment]
> From that logic (have the contacts available but don’t have them clutter up your Contacts & synced) our current approach with the separate DB is fine.

What will trigger the insert of an address into the db? On loading the message list? Opening one message?

**Comment by ChristophWurst** [comment]
> so that we can use the collected addresses on mobile as well

Simply use the mail app on your phone and you'll have the auto collected addresses too ;-)

**Comment by jancborchardt** [comment]
> What will trigger the insert of an address into the db? On loading the message list? Opening one message?

@DeepDiver1975 I’d say for now having sent an email to that person. I think that’s how Gmail does it. Otherwise you will end up with lots of bullshit addresses just cause you opened messages.
